Gems/EMotionFX/Code/EMotionFX/Source/AnimGraphMotionNode.cpp

```diff
@@ -664,7 +664,16 @@ namespace EMotionFX
                 const float randomRange = previousIndexCumulativeWeight + m_motionRandomSelectionCumulativeWeights.back().second - currentIndexCumulativeWeight;
 
                 // Picking a random number between [0, randomRange)
-                const float randomValue = animGraphInstance->GetLcgRandom().GetRandomFloat() * randomRange;
+                float randomValue = 0.0f;
+                if (animGraphInstance->IsNetworkEnabled())
+                {
+                    // Using a seeded random in order to generate predictable result in network. 
+                    randomValue = animGraphInstance->GetLcgRandom().GetRandomFloat() * randomRange;
+                }
+                else
+                {
+                    randomValue = MCore::Random::RandF(0, randomRange);
+                }
```

Comment:
This only fixes the issue for offline mode. Can't we increment the seed per actor instance on a deterministic way and thus get different and actually random results per actor instance?

So basically moving the start seed into the emfx manager or something and with each actor or anim graph instance we increase that seed value to still be deterministic but use different seeds per instance.

Reply:
+1.  I'm a big fan of deterministic behavior by default, _especially_ for things that can affect gameplay, whether offline or online.  From a quick look at the code, it seems like the issue is that the seed is always initialized to 1234 right now. Maybe on instance copies, set the new instance's seed to the previous instance's seed + 1?  It can still _become_ non-deterministic if the instance copy order is dependent on time, load order, etc, but it seems like an improvement over guaranteed non-determinism.

Reply:
Seems like most people like to keep it deterministic online and offline, so I will make a change here as Ben suggested